### PR TITLE
resolve failing cases

### PR DESCRIPTION
**Pull request by ShubhamRasal**

### CODE REVIEW

**Comment by ShubhamRasal** [comment]
crtsh and commoncrawl errors are very random. 
most of the time tests fails due to 503, and connection timeout error of sources.

**Review by Mzack9999 — CHANGES_REQUESTED** [review]
lgtm - not related but I'd like to propose removing `push` from github actions:
```
on:
  push:
  pull_request:
```
The interested actions are:
- [x] build-test.yml
- [ ] lint-test.yml

**Review by Mzack9999 — APPROVED** [review]
lgtm - faulty sources removal tracked at https://github.com/projectdiscovery/subfinder/issues/722

**Review by ehsandeep — APPROVED** [review]

File: v2/pkg/subscraping/agent.go [hunk 1]
```diff
@@ -24,6 +25,9 @@ func NewSession(domain string, proxy string, rateLimit, timeout int) (*Session,
 		TLSClientConfig: &tls.Config{
 			InsecureSkipVerify: true,
 		},
+		Dial: (&net.Dialer{
+			Timeout: time.Duration(timeout) * time.Second,
+		}).Dial,
```
Comment (vzamanillo) [inline v2/pkg/subscraping/agent.go:30]: The timeout is set when the http.client is initialized, this is a dup.

**Pull request merged**